src/gda/backend_gda.cpp

```diff
@@ -910,6 +914,74 @@ void GDABackend::create_queues() {
 
   create_cqs(ncqes);
   create_qps(sq_size);
+
+  map_ctxs_to_qps();
+}
+
+void GDABackend::map_ctxs_to_qps() {
+  std::map<int, std::set<int>> port_idxs;
+  int new_qp_idx;
+  int cur_qp_idx;
+  int ctx_port;
+
+  /* We can't remap anything */
+  if (maximum_num_contexts_ == 1) {
+    return;
+  }
+
+#ifdef GDA_BNXT
+  if (map_ctx_to_port) {
+    /* If we assume two PEs and a default context and two user context,
+     * initially QPs are in the following port order:
+     *
+     * Labels :| DCTX PE0 | DCTX PE1 | CTX0 PE0 | CTX0 PE1 | CTX1 PE0 | CTX1 PE1 |
+     * QPs    :| QP0      | QP1      | QP2      | QP3      | QP4      | QP5      |
+     * Port   :| 0        | 1        | 0        | 1        | 0        | 1        |
+     *
+     * This creates the pattern where PE1 is always mapped to port 0 but we want it
+     * to use both ports to maximize throughput/bandwidth.
+     *
+     * So we reorder our QPs
+     *
+     * Labels :| DCTX PE0 | DCTX PE1 | CTX0 PE0 | CTX0 PE1 | CTX1 PE0 | CTX1 PE1 |
+     * QPs    :| QP0      | QP1      | QP2      | QP4      | QP3      | QP5      |
+     * Port   :| 0        | 1        | 0        | 0        | 1        | 1        |
+     *
+     * The default context will remain unchanged but the user contexts will each
```

Comment:
because the solution doesn't fix default context it will not resolve perf issue in deepEP right now.